### DISCUSSION TITLE
Usage with prerender.io (redirect certain user agents to a different url)

### DISCUSSION THREAD

**Discussion by oogxdd** [vercel/community#53] [Help] [answered]
Hi. I'm building SPA using React, i need to support Opengraph tags (based on dynamic content) for social sharing

I'm using prerender.io to redirect requests from certain user agents (`'googlebot', 'twitterbot', linkedinbot', etc`) to a prerender.io website where my page gets built and then returned to these workers)

I host my website on AWS EC2 instance and use nginx to proxy requests from bots to service.prerender.io. (https://medium.com/bleeding-edge/it-can-work-easily-serverless-react-and-web-crawlers-7e2afbc39590 here described how its done at the end of the article)

My question is: can I achieve same effect if I keep my website being hosted on Vercel? Can i set up something like nginx, proxy settings based on which headers i receive)?

ps. i can't use solution like nextjs because i an issue with their navigation https://github.com/vercel/next.js/issues/17660

**Comment by kaioken** [comment]
@maximignatev hey, were you able to get this resolved? running the site on vercel with prerender.io?

Thanks

**Comment by oogxdd** [comment]
@kaioken hey no i went with EC2 instance

**Comment by leerob** [comment, marked as answer]
With Middleware and Next.js 12, you can now easily redirect certain user agents to different URLs. 


https://nextjs.org/docs/middleware

**Comment by kg-currenxie** [comment]
Any solutions without nextjs?

**Reply by amyegan** [reply]
[Edge Middleware](https://vercel.com/docs/concepts/functions/edge-middleware) is available for other frameworks on Vercel. The new [Build Output API](https://vercel.com/blog/build-output-api) allows any framework to use Edge Middleware and other Vercel infrastructure building blocks. More details here: https://vercel.com/docs/build-output-api/v3
